Compiler errors while building on centos7 image

Hi, We are trying to build tensonflow/io v0.10.0. We are using a centos7 base instead of an ubuntu base, but otherwise our build process is similar:
```bash
# Build and run the Docker image
docker build -f tools/dev/Dockerfile -t tfio-dev .
# Run the docker image
docker run --rm --net=host -v ${PWD}:/v -w /v tfio-dev
# Inside the container configure install
./configure.sh
# Build TensorFlow I/O. More info on flags https://www.tensorflow.org/install/source#configuration_options
bazel build -c opt --copt=-fPIC -s --verbose_failures //tensorflow_io/core/...
# Run c++ tests
bazel test --verbose_failures //tensorflow_io/avro/... --action_env=LD_LIBRARY_PATH=$LD_LIBRARY_PATH
# Build the TensorFlow I/O package
python setup.py bdist_wheel
# TODO Install the wheel
pip install --upgrade --force-reinstall dist/tensorflow_io-*.whl
# Run tests
pytest -s -v tensorflow_io/avro/python/tests
```
Here's docker file: 
```bash
FROM ${private-registry}/tensorflow/tensorflow-builder:0.0.1
RUN yum install -y \
    git \
    gcc \
    g++ \
    gdb \
    make \
    patch \
    curl \
    nano \
    unzip \
    ffmpeg \
    dnsutils

ARG BAZEL_VERSION=0.29.0
ARG BAZEL_OS=linux
RUN curl -sL https://github.com/bazelbuild/bazel/releases/download/${BAZEL_VERSION}/bazel-${BAZEL_VERSION}-installer-${BAZEL_OS}-x86_64.sh -o bazel-install.sh && \
    bash -x bazel-install.sh && \
    rm bazel-install.sh
ARG CONDA_OS=Linux
# Miniconda - Python 3.6, 64-bit, x86, latest
RUN curl -sL https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -o mconda-install.sh && \
    bash -x mconda-install.sh -b -p miniconda && \
    rm mconda-install.sh
ENV PATH="/miniconda/bin:$PATH"
ARG CONDA_ADD_PACKAGES=""
RUN conda create -y -q -n tfio-dev python=3.6 ${CONDA_ADD_PACKAGES}

ARG PIP_ADD_PACKAGES=""
ARG ARROW_VERSION=0.14.1
RUN /bin/bash -c "source activate tfio-dev && python -m pip install \
    avro-python3 \
    pytest \
    pylint \
    boto3 \
    google-cloud-pubsub==0.39.1 \
    pyarrow==${ARROW_VERSION} \
    pandas \
    fastavro \
    gast==0.2.2 \
    ${PIP_ADD_PACKAGES} \
    "
ENV TFIO_DATAPATH=bazel-bin
```
The error we get is attached.
[kafka-err.txt](https://github.com/tensorflow/io/files/4046375/kafka-err.txt)

GCC 4.8 and libstdc++.so on CentOS 7 may not be possible initally. I tried to build with Centos 7 and devtoolset 7, was close but get the following:
```
bazel-out/host/bin/external/com_github_google_flatbuffers/_objs/flatc/idl_gen_js_ts.o:external/com_github_google_flatbuffers/src/idl_gen_js_ts.cpp:function flatbuffers::jsts::JsTsGenerator::GenDocComment(std::vector<std::string, std::allocator<std::string> > const&, std::string*, std::string const&, char const*): error: undefined reference to 'std::__throw_out_of_range_fmt(char const*, ...)'
```

Looks like the libstdc++.so on CentOS 7 will need to be updated to get around this issue.

--- a/.github/workflows/build.develop.py
+++ b/.github/workflows/build.develop.py
@@ -1,0 +1,17 @@
+import sys
+
+source = sys.argv[1]
+section = sys.argv[2]
+with open (source, "r") as f:
+    lines = [line.rstrip() for line in list(f)]
+
+# Remove lines before section title
+lines = lines[lines.index(section):]
+
+# Remove lines outside (including) "```sh" and "```"
+lines = lines[lines.index("```sh")+1:lines.index("```")]
+
+# Remove sudo
+lines = [(line[len("sudo "):] if line.startswith("sudo ") else line) for line in lines]
+
+print("\n".join(lines))

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -27,6 +27,30 @@ jobs:
           sudo bash -e bazel-${BAZEL_VERSION}-installer-${BAZEL_OS}-x86_64.sh
           bazel run -s --verbose_failures //tools/lint:check
 
+  ubuntu-1804:
+    name: Ubuntu 18.04
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - run: |
+          set -x -e
+          python3 .github/workflows/build.develop.py README.md "##### Ubuntu 18.04" > source.sh
+          cat source.sh
+          docker run -i --rm -v $PWD:/v -w /v --net=host ubuntu:18.04 \
+            bash -x -e source.sh
+
+  centos-7:
+    name: CentOS 7
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - run: |
+          set -x -e
+          python3 .github/workflows/build.develop.py README.md "##### CentOS 7" > source.sh
+          cat source.sh
+          docker run -i --rm -v $PWD:/v -w /v --net=host centos:7 \
+            bash -x -e source.sh
+
   macos-bazel:
     name: Bazel macOS
     runs-on: macos-latest

--- a/README.md
+++ b/README.md
@@ -143,6 +143,82 @@ available, it is possible to run tests with pytest, e.g.:
 
 #### Linux
 
+Development of tensorflow-io on Linux is similiar to development on macOS. The required packages
+are gcc, g++, git, bazel, and python 3. Newer versions of gcc or python than default system installed
+versions might be required though.
+
+##### Ubuntu 18.04
+
+Ubuntu 18.04 requires gcc/g++, git, and python 3. However, due to a dependency of grpc, python-dev
+is also needed. As such the following will install dependencies and build the shared libraries on
+Ubuntu 18.04:
+```sh
+# Install gcc/g++, git, unzip/curl (for bazel), and python3
+sudo apt-get -y -qq update
+sudo apt-get -y -qq install gcc g++ git unzip curl python3-pip python-dev
+
+# Install Bazel 2.0.0
+curl -sSOL https://github.com/bazelbuild/bazel/releases/download/2.0.0/bazel-2.0.0-installer-linux-x86_64.sh
+sudo bash -x -e bazel-2.0.0-installer-linux-x86_64.sh
+
+# Upgrade pip
+python3 -m pip install -U pip
+
+# Install tensorflow and configure bazel with rh-python36
+./configure.sh
+
+# Build shared libraries
+bazel build -s --verbose_failures //tensorflow_io/...
+
+# Once build is complete, shared libraries will be available in
+# `bazel-bin/tensorflow_io/core/python/ops/` and it is possible
+# to run tests with `pytest`, e.g.:
+python3 -m pip install pytest
+TFIO_DATAPATH=bazel-bin python3 -m pytest -s -v tests/test_serialization_eager.py
+```
+
+##### CentOS 7
+
+On CentOS 7, the default python and gcc version are too old to build tensorflow-io's shared
+libraries (.so). The gcc provided by Developer Toolset and rh-python36 should be used instead.
+Also, the libstdc++ has to be linked statically to avoid discrepancy of libstdc++ installed on
+CentOS vs. newer gcc version by devtoolset.
+
+The following will install bazel, devtoolset-9, rh-python36, and build the shared libraries:
+```sh
+# Install centos-release-scl, then install gcc/g++ (devtoolset), git, and python 3
+sudo yum install -y centos-release-scl
+sudo yum install -y devtoolset-9 git rh-python36
+
+# Install Bazel 2.0.0
+curl -sSOL https://github.com/bazelbuild/bazel/releases/download/2.0.0/bazel-2.0.0-installer-linux-x86_64.sh
+sudo bash -x -e bazel-2.0.0-installer-linux-x86_64.sh
+
+# Upgrade pip
+scl enable rh-python36 devtoolset-9 \
+    'python3 -m pip install -U pip'
+
+# Install tensorflow and configure bazel with rh-python36
+scl enable rh-python36 devtoolset-9 \
+    './configure.sh'
+
+# Build shared libraries
+BAZEL_LINKOPTS="-static-libstdc++ -static-libgcc" BAZEL_LINKLIBS="-lm -l%:libstdc++.a" \
+  scl enable rh-python36 devtoolset-9 \
+    'bazel build -s --verbose_failures //tensorflow_io/...'
+
+# Once build is complete, shared libraries will be available in
+# `bazel-bin/tensorflow_io/core/python/ops/` and it is possible
+# to run tests with `pytest`, e.g.:
+scl enable rh-python36 devtoolset-9 \
+    'python3 -m pip install pytest'
+TFIO_DATAPATH=bazel-bin \
+  scl enable rh-python36 devtoolset-9 \
+    'python3 -m pytest -s -v tests/test_serialization_eager.py'
+```
+
+#### Docker
+
 For Python development, a reference Dockerfile [here](tools/dev/Dockerfile) can be
 used to build the TensorFlow I/O package (`tensorflow-io`) from source:
 ```sh

